src/Compiler/Checking/Expressions/CheckExpressions.fs

```diff
@@ -9270,6 +9270,13 @@ and TcValueItemThen cenv overallTy env vref tpenv mItem afterResolution delayed
                 let _tpR, tpenv = TcTypeOrMeasureParameter None cenv env ImplicitlyBoundTyparsAllowed.NoNewTypars tpenv tp
                 let vExpr = TcNameOfExprResult cenv id mExprAndTypeArgs
                 let vexpFlex = MakeApplicableExprNoFlex cenv vExpr
+
+                // Record the resolution of the `nameof` usage so that we can classify it correctly later.
+                do
+                    match afterResolution with
+                    | AfterResolution.RecordResolution (_, callSink, _, _) -> callSink emptyTyparInst
+                    | AfterResolution.DoNothing -> ()
+
                 PropagateThenTcDelayed cenv overallTy env tpenv mExprAndTypeArgs vexpFlex g.string_ty ExprAtomicFlag.Atomic otherDelayed
             | _ ->
                 error (Error(FSComp.SR.expressionHasNoName(), mExprAndTypeArgs))
```

Comment:
Should `nameof` have keyword coloring in scenarios like these?

```fsharp
let f (x : 'a) (y : 'b) = nameof<'c> // FS0039: The type parameter 'c is not defined.
let bar = nameof<int> // FS3250: Expression does not have a name.
```

If so, we should call the name resolution sink unconditionally before this `match`:

https://github.com/dotnet/fsharp/blob/74eea31868152c5b66827288104c8204181c1657/src/Compiler/Checking/Expressions/CheckExpressions.fs#L9267-L9268

Reply:
I think it probably should, so: 3f91e70